network/src/network.rs

```diff
@@ -1026,10 +1047,10 @@ impl NetworkService {
         }
     }
 
-    pub fn start<S: ToString>(
+    pub fn start<NV: ToString, TN: ToString>(
```

Comment:
It's not necessary to use generic here. It's simpler to just ask for callee to pass in String.

Reply:
I found this parameter is unnecessary, removed it in new commit, please review again, thanks